### ISSUE TITLE
How do I debug the dev version?

### ISSUE THREAD

**Issue by mdolnik** [dresden-elektronik/deconz-rest-plugin#2351]
Hi there,

I have gotten the dev version of the rest-plugin working in ubuntu by using the [instructions on the home page](https://github.com/dresden-elektronik/deconz-rest-plugin#install-deconz-development-package-optional-linux-only) and it seems to compile properly as far as I can tell.

My question is what do I do next?

I am able to get Deconz outputting logs by running `deCONZ --dbg-info=2` and I am able to get my partially working [friends of hue switch (illumra)](https://github.com/dresden-elektronik/deconz-rest-plugin/issues/2108) to output events with logs such as
```
20:05:05:803 Websocket 192.168.1.70:32966 send message: {"e":"changed","id":"2","r":"sensors","state":{"buttonevent":20,"lastupdated":"2020-01-21T04:05:05"},"t":"event","uniqueid":"00:00:00:00:01:50:58:d2-f2"} (ret = 153)
```

But when I modify the code which prints this log (at least what I think prints the log) by modifying `WebSocketServer::broadcastTextMessage()` in `websocket_server.cpp` to change the log to something like `...send MY MODIFIED message...` and when I recompile and copy `libde_rest_plugin.so` to `/usr/share/deCONZ/plugins` and I re-run `deCONZ --dbg-info=2` the log text isn't updated and still says `send message`

Is there something I'm missing here?

Is there an alternate means to confirm that I have modified and compiled and updated the plugin correctly?

**Comment by ebaauw** [comment]
Did you remove the stock rest api plugin from the `plugins` directory?  If you only rename it, it will still be loaded.

**Comment by mdolnik** [comment]
After compiling I ran the `sudo cp ../libde_rest_plugin.so /usr/share/deCONZ/plugins` command which I've confirmed overwrites the `libde_rest_plugin.so` file in the plugins directory.

Is that what you mean by removing the plugin? Or is there another plugin file I need to remove?

**Comment by ebaauw** [comment]
No, using `sudo cp` is good.

**Comment by mdolnik** [comment]
I am brand new to the C++ and compiling world, so I can't say I know too much of what's going on, but I managed to get it working.

I tried running `make clean` then I ran `qmake && make -j2` and when I re-copied the modified `libde_rest_plugin.so` to `/usr/share/deCONZ/plugins` and ran `deCONZ --dbg-info=2` I am now getting the updated logging messages that confirm that my modified code is being used.

So `make clean` is what solved my issue, and I only needed to run it once. I'm not entirely sure why it worked, as looking into that command people seem to think it's not necessary, but it worked for me.
